apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -444,11 +454,28 @@ private static ConstantCallSite internalBootstrap(MethodHandles.Lookup lookup, S
         }
     }
 
-    private static List<String> getClassNamesFromBundledPlugin(String adviceClassName, ClassLoader adviceClassLoader) throws IOException, URISyntaxException {
-        if (!adviceClassName.startsWith(EMBEDDED_PLUGINS_PACKAGE_PREFIX)) {
-            throw new IllegalArgumentException("invalid advice class location : " + adviceClassName);
+    private static boolean addRequiredModuleOpens(Map<String, List<String>> requiredModuleOpens, ClassLoader targetClassLoader, ClassLoader pluginClassLoader) {
```

Comment:
javadoc required, especially what the return value means.
I assume you'll find it difficult to explain, which indicates it is not very clear success/failure. I propose to either change what the method does, so that the return value represents success vs. error, or not returning a value and relying on an exception instead.

Sorry for nitpicking, this code is already complex as it is....